app/Global.scala

```diff
@@ -1,24 +1,19 @@
-import com.google.inject.{Guice, AbstractModule}
-import play.api.GlobalSettings
-import services.{SimpleUUIDGenerator, UUIDGenerator}
-
-/**
- * Set up the Guice injector and provide the mechanism for return objects from the dependency graph.
- */
-object Global extends GlobalSettings {
-
-  /**
-   * Bind types such that whenever UUIDGenerator is required, an instance of SimpleUUIDGenerator will be used.
-   */
-  val injector = Guice.createInjector(new AbstractModule {
-    protected def configure() {
-      bind(classOf[UUIDGenerator]).to(classOf[SimpleUUIDGenerator])
-    }
-  })
-
-  /**
-   * Controllers must be resolved through the application context. There is a special method of GlobalSettings
-   * that we can override to resolve a given controller. This resolution is required by the Play router.
-   */
-  override def getControllerInstance[A](controllerClass: Class[A]): A = injector.getInstance(controllerClass)
-}
+import com.google.inject.Guice
+import play.api._
+
+/**
+ * Set up the Guice injector and provide the mechanism for return objects from the dependency graph.
+ */
+object Global extends GlobalSettings {
+  private lazy val injector = {
+//    Play.isProd match {
+//      case true => Guice.createInjector(new Module)
+//      case false => Guice.createInjector(new Module)
+//    }
+    Guice.createInjector(new Module)

```

Comment:
DI complicates code, use it judiciously. Class parameters are a way to achieve this.